rllib/env/multi_agent_env.py

```diff
@@ -555,6 +555,21 @@ def try_reset(self, env_id: Optional[EnvID] = None) -> Optional[MultiEnvDict]:
             ret[idx] = obs
         return ret
 
+    @override(BaseEnv)
+    def try_restart(self, env_id: Optional[EnvID] = None) -> None:
+        if isinstance(env_id, int):
+            env_id = [env_id]
+        if env_id is None:
+            env_id = list(range(len(self.envs)))
+        for idx in env_id:
+            # Recreate the sub-env.
+            self.envs[idx] = self.make_env(idx)
+            # Replace the multi-agent env state at the index.
+            self.env_states[idx] = _MultiAgentEnvState(self.envs[idx])
+            # Remove done flag at index.
+            if idx in self.dones:
+                self.dones.remove(idx)
```

Comment:
how are these data bits initialized in the first place?
should we create a util func MultiAgentEnvWrapper.init_env_state(idx), which does these two operations, and gets called here.
then in the constructor, we simplly do: ```for i in range(len(self.envs)): self.init_env_state(i)```

Reply:
done